lib/elixir/lib/string.ex

```diff
@@ -3066,74 +3066,15 @@ defmodule String do
   @spec jaro_distance(t, t) :: float
   def jaro_distance(string1, string2)
 
-  def jaro_distance(string, string), do: 1.0
+  def jaro_distance(string, string) when is_binary(string), do: 1.0
   def jaro_distance(_string, ""), do: 0.0
   def jaro_distance("", _string), do: 0.0
```

Comment:
I felt we could keep these optimizations since they weren't in the Erlang version, WDYT?

Reply:
Fine to keep them, yeah. :)